pkg/cli/admin/release/new.go

```diff
@@ -1391,17 +1394,22 @@ func writePayload(w io.Writer, is *imageapi.ImageStream, cm *CincinnatiMetadata,
 	if err := iterateExtractedManifests(ordered, metadata, func(directory string, contents []os.FileInfo, operator string) error {
 		transform := NopManifestMapper
 
-		if fi := takeFileByName(&contents, "image-references"); fi != nil {
-			path := filepath.Join(directory, fi.Name())
-			klog.V(2).Infof("Perform image replacement based on inclusion of %s", path)
-			transform, err = NewTransformFromImageStreamFile(path, is, allowMissingImages, errOut)
-			if err != nil {
-				return fmt.Errorf("operator %q contained an invalid image-references file: %s", operator, err)
+		// If there is an image-references file in the directory, we will need to replace image references in the manifests
+		// We could iterate over contents to detect the file, but it is easier to just directly check
+		imageReferencesPath := filepath.Join(directory, imageReferencesImageStreamFilename)
+		if _, err := os.Stat(imageReferencesPath); err == nil {
+			klog.V(2).Infof("Perform image replacement based on inclusion of %s", imageReferencesImageStreamFilename)
+			t, nErr := NewTransformFromImageStreamFile(imageReferencesPath, is, allowMissingImages, errOut)
+			if nErr != nil {
```

Comment:
I am not that aware of the code context. I am checking for any potential regressions. Previously, the `takeFileByName` would return either `nil` or the `image-references` file.

The new changes are able to pass an `image-references` *directory* if present to the following logic, which would probably cause the code to fail. Can such a directory be present?

Reply:
Previously, such a directory would get silently skipped. With the new changes, it seems like this would cause the logic to fail. I am not sure what is a better option.

Reply:
Good point, thanks. I actually like the change less after the delay. I pivoted the code back to iterating `contents` while still getting rid of the side-effect.